### DISCUSSION TITLE
文档上似乎没有提及如何监听Group节点的展开与收起状态的变化

### DISCUSSION THREAD

**Discussion by cpf23333** [didi/LogicFlow#1290] [Q&A] [answered]
节点展开或收起后需要更改一些数据

目前通过重写foldGroup方法实现了对状态的监测，但这样是不是不太优雅，也可能会出现问题？

可不可以在GroupNode.model中提供方法，在展开收起状态变化后触发？

**Comment by wumail** [comment]
目前看来重写foldGroup不会有什么问题，也可以重写view上的getFoldIcon方法（这样可能更好）
<image src='https://github.com/didi/LogicFlow/assets/56008486/2b0d9a1c-eb4c-4c66-bdb7-d3d9d9d56ac1' width='50%'/>
即使内置折叠的触发，也是放在这两个地方

**Reply by cpf23333** [reply]
刚刚测试了一下，重写getFoldIcon不可行

这个方法在鼠标移入移出group对应节点时也会触发，造成不必要的性能消耗

看起来只能采用重写foldGroup的方案了

**Reply by wumail** [reply]
写在onClick里，相对麻烦，但是感觉比重写foldGroup合理一点

**Reply by cpf23333** [reply]
是指缩放按钮的点击事件吗

另外我在https://github.com/didi/LogicFlow/blob/f8e51cc53d9b465ed868707cdd4e7f8f47a2a9d9/packages/extension/examples/group/index.mjs#L46C19-L46C19

找到了和我的需求类似的demo，也是重写foldGroup

**Reply by wumail** [reply, marked as answer]
是的，重写foldGroup是OK的，只是感觉上在点击事件回调里面触发监听看起来会更合理一点

**Reply by cpf23333** [reply]
好的，那目前就重写foldGroup，如果后续有变动再看情况更改
